### ISSUE TITLE
NewsFlash core dump while trying to update feeds

### ISSUE THREAD

**Issue by xnrvi** [flathub/io.gitlab.news_flash.NewsFlash#9]
NewsFlash repeatedly dumps core when trying to update the feeds I fed it. I attached the error log. Can you fix this? Thanks.

[newsflash-fault.txt](https://github.com/flathub/io.gitlab.news_flash.NewsFlash/files/14229721/newsflash-fault.txt)

**Comment by jangernert** [comment]
Can you export an OPML file and upload it here? I need to figure out  which articles are triggering the segfault.

**Comment by xnrvi** [comment]
Thanks for the reply. Here you go and good luck bug hunting.
[NewsFlash.OPML.zip](https://github.com/flathub/io.gitlab.news_flash.NewsFlash/files/14251195/NewsFlash.OPML.zip)

**Comment by jangernert** [comment]
I did import your OPML and have been refreshing the feeds a few times since yesterday evening. I did however not hit a segfault yet. Does it happen every time for you?

Your OPML did encourage me to commit some improvements to the downloading of feeds for the local RSS backend though. So thats nice at least.

**Comment by xnrvi** [comment]
Hi again. My estimate is that an automated refresh will cause a segfault approximately 6-7 out of 10 times. A manual refresh may result in a crash too. If I configure the program to update feeds at startup this will almost always result in a crash. When a crash is imminent the feed update icon (of the middle pane) may look distorted or without animation. Hope this helps, but I know how challenging it can be to find these things.

**Comment by jangernert** [comment]
I added some [null checks](https://gitlab.com/news-flash/article_scraper/-/commit/b13673ce3b94b8c2810573bf61dfc8f0724b63df) before unlinking nodes with libxml2. Maybe it helps. I still haven't had a single segfault with your feed list.

It  would be cool if  you could install this CI build (`flatpak install`) and see if the issue was fixed. This will not overwrite your existing installation. Instead you'll have a second development version on your system, with completely separate configuration and data.

https://gitlab.com/news-flash/news_flash_gtk/-/jobs/6165277037/artifacts/raw/io.gitlab.news_flash.NewsFlash.Devel-3.1.0-2b948c98.flatpak

**Comment by xnrvi** [comment]
This seems to infect my arch/manjaro/cinnamon installation with 'gnomes' again. I have never considered gnomes or other woodland creatures to be much of a space-faring subspecies of human scouts, but ... I'll give it a shot. Hang on. There:

20:48:33 - INFO  - NewsFlash 3.1.0-devel (io.gitlab.news_flash.NewsFlash.Devel) (news_flash_gtk::app:92)
20:48:33 - WARN  - No account configured               (news_flash_gtk::app:103)
20:48:34 - INFO  - WebKit version: 2.42.5              (news_flash_gtk::article_view::imp:141)
flatpak-spawn: Invalid byte sequence in conversion input
Try "flatpak-spawn --help" for more information.
20:48:34 - INFO  - Networkmonitor connectivity: Local (available false) (news_flash_gtk::util:115)
20:48:34 - INFO  - pinging 'http://example.com/' succeeded (news_flash_gtk::util:127)
20:48:35 - INFO  - Can't set offline false because not logged in yet (news_flash_gtk::app:2542)

?

**Comment by jangernert** [comment]
Ah, that's when the necessary locale is not installed and webkit tries to spawn a new process. You can simply override the locale since this is only for testing purposes

```
flatpak run --env=LC_ALL="en_US.UTF-8" io.gitlab.news_flash.NewsFlash.Devel

```

**Comment by jangernert** [comment]
> src/nv_gbm.c:242: GBM-DRV error (nv_gbm_create_device_native): nv_common_gbm_create_device failed (ret=-1)

This is most likely something related to webkit and their rendering in conjunction with nvidia. But that shouldn't really matter for now.

> Failed to set background message: This interface requires version 2, but 1 is available 

That's interesting, I wasn't aware there were multiple versions of this portal. Its basically looking for this xdg-portal implementation on your system and not finding anything: https://github.com/flatpak/xdg-desktop-portal/blob/main/data/org.freedesktop.portal.Background.xml

> Message: already borrowed: BorrowMutError
Location: ../src/app.rs:1011

This is something inside the codebase of Newsflash and also something I wasn't aware of. I know all this is not solving your initial problem, but its great for me discovering all these smaller issues while trying to figure out the segfault :D 

> That does seem to work better at first sight (!)

Fingers crossed. Maybe its fixed. Or maybe that one problematic article was already pushed out of the feed xml by newer articles.

**Comment by xnrvi** [comment]
Indeed. It looks like the 'Business Insider' feed was the source of the problem. I removed it from the feeds and now the 'official' manjaro repo version works again too. Thanks for your efforts!

**Comment by jangernert** [comment]
did the development version crash again?

**Comment by jangernert** [comment]
I  just checked the OPML: and there is no "Business Insider" feed, only "All Insider Content". Did you mean that?

**Comment by xnrvi** [comment]
That one, yes. It used to be Business Insider. The repo version works better without that particular feed, so maybe it boils down to the format / content of certain feeds.

**Comment by jangernert** [comment]
> The repo version works better without that particular feed, so maybe it boils down to the format / content of certain feeds.

Yes, the content of feeds is processed with libxml2 during the sync (stripping out scripts, looking for a suitable thumbnail, general cleanup, etc). It would be interesting if the changes in the development build actually fix the crash. I know its hard to verify since the issue is not easy to reliably reproduce.

**Comment by jangernert** [comment]
Thanks, I was just making sure. Sometimes small misunderstandings can happen when things are not spelled out 100%.

I think we can close this issue then. And you can let me know if the crash still appears after the next update :)
